_posts/2020-10-16-how-we-estimate-rails-application-size.md

```diff
@@ -0,0 +1,176 @@
+changing which might be an issue if you want to upgrade from Rails 6.0 to
+6.1.
+
+### Codebase (Lines of Code)
+
+In order to calculate the lines of code in a codebase, I prefer
+[`rails_stats`](https://github.com/bleonard/rails_stats) which enhances Rails's
+native `rake stats`.
+
+One of the things I like about it is that `rails_stats` doesn't need to load the
+Rails environment in order to calculate all statistics. Sometimes that can be
+annoying, especially when you are trying to size up a really, really old
+application which requires a very specific set of dependencies (think a very
+specific version of Ruby, Rails, Bundler, and Rubygems)
+
+You will need a `Rakefile` in the directory where you call `rake stats` and you
+will need to require `rails_stats` within that file:
+
+```ruby
+# Rakefile
+require "rails_stats"
+```
+
+Then you can call it:
+
+```bash
+$ rake stats\[./dash\]
+
+Directory: /Users/etagwerker/Projects/fastruby/dash
+
++----------------------+-------+-------+---------+---------+-----+-------+
+| Name                 | Lines |   LOC | Classes | Methods | M/C | LOC/M |
++----------------------+-------+-------+---------+---------+-----+-------+
+| Mailers              |     4 |     4 |       1 |       0 |   0 |     0 |
+| Models               |   257 |   211 |      10 |      27 |   2 |     5 |
+| Workers              |   108 |    84 |       3 |      10 |   3 |     6 |
+| Javascripts          |  1487 |  1462 |       0 |     188 |   0 |     5 |
+| Jobs                 |     7 |     2 |       1 |       0 |   0 |     0 |
+| Controllers          |    98 |    85 |       6 |      13 |   2 |     4 |
+| Helpers              |    21 |    18 |       0 |       4 |   0 |     2 |
+| Services             |    31 |    25 |       1 |       6 |   6 |     2 |
+| Channels             |     8 |     8 |       2 |       0 |   0 |     0 |
+| Configuration        |   456 |   139 |       1 |       0 |   0 |     0 |
+| Spec Support         |   259 |   108 |       0 |       0 |   0 |     0 |
+| Feature Tests        |    47 |    40 |       0 |       0 |   0 |     0 |
+| Model Tests          |   122 |    97 |       0 |       0 |   0 |     0 |
+| Worker Tests         |    77 |    53 |       0 |       0 |   0 |     0 |
+| Controller Tests     |   180 |   144 |       0 |       0 |   0 |     0 |
+| Other Tests          |    55 |    45 |       0 |       1 |   0 |    43 |
+| Service Tests        |    33 |     0 |       0 |       0 |   0 |     0 |
++----------------------+-------+-------+---------+---------+-----+-------+
+| Total                |  3250 |  2525 |      25 |     249 |   9 |     8 |
++----------------------+-------+-------+---------+---------+-----+-------+
+  Code LOC: 2038     Test LOC: 487     Code to Test Ratio: 1:0.2
+```
+
+You now know that this project has **2,038 lines of code** (both Ruby and
+JavaScript files) in the application directory and **487 lines of code** in
+its test directory.
+
```

Comment:
@etagwerker Maybe here you can mention that dash is a tiny/small app, since at the beginning you say "tiny, medium, or huge Rails applications" but you don't mention any numbers.